targets/XPlatCppSdk/source/code/source/playfab/PlayFabAndroidHttpPlugin.cpp

```diff
@@ -0,0 +1,743 @@
+#include <stdafx.h>
+
+#include <jni.h>
+#include <android/log.h>
+
+#include <playfab/PlayFabAndroidHttpPlugin.h>
+#include <playfab/PlayFabSettings.h>
+#include <stdexcept>
+#include <vector>
+
+namespace PlayFab
+{
+    struct PlayFabAndroidHelper
```

Comment:
What's the reason for using a struct here instead of a class ? Was this a pattern you were following from other plugin examples ? Usually struct is a container for POD types.

Reply:
The reason of using structure are 
1. This is internal data structure which is used by PlayFabAndroidHttpPlugin, so no need encapsulating because this members should be fully accessed.
2. It doesn't need inheritance. And in this pimpl design pattern is usually used straightforward. If it needs to use inheritance this should be used different design pattern.
3. Structure and Class are exactly the same structurally.

---
In reply to: [274670039](https://github.com/PlayFab/SDKGenerator/pull/470#discussion_r274670039) [](ancestors = 274670039)